lib/testsuite/index.js

```diff
@@ -838,7 +845,36 @@ class TestSuite {
 
     return runnable.run(this.commandQueue);
   }
-  
+
+  takeSnapshot(node) {
+    const commands = this.reporter.currentSection.commands || [];
+    const currentCommand = commands[commands.length-1];
+
+    if (this.settings.trace.enabled && node.isTraceable) {
+      const snapShotPath = Snapshots.getFileName({
+        testSuite: this.api.currentTest.module,
+        commandName: node.fullName,
+        traceSettings: this.settings.trace,
+        output_folder: this.settings.output_folder
+      });
+
+      this.__snapShot = new Promise((resolve, reject) => {
+        this.api.saveSnapshot(snapShotPath, (result => {
+          if (currentCommand) {
+            currentCommand.domSnapshot = result;
+          }
+          resolve(result);
+        }));
+      });
+    } else {
+      if (currentCommand && this.__snapShot) {
+        this.__snapShot.then(prevSnapshot => {
+          currentCommand.domSnapshot = prevSnapshot;
+        });
+      }
```

Comment:
shoudln't we consider the case where `this.api.saveSnapshot` results into failure

Reply:
I think if `snapShot` fails with an error that is not handled

Reply:
The catch block at [this line](https://github.com/nightwatchjs/nightwatch/pull/3508/files/4e4cde30c8f66d79386049162fe9f5fefdc8ae57#diff-67f2d140983390c2d6f180133d070a60fd9dd5370bf5cef803065ea6e5651706R56) should handle all exceptions.

Reply:
It sends error in the callback, it might end up in `result` of the saveSnapshot callback function

Reply:
Yes, and that is fine I think. As long as it does not throw and unhandled exception. Users can see the error in the result object.